src/sources/demo_logs.rs

```diff
@@ -171,11 +179,12 @@ async fn demo_logs_source(
                         log.try_insert(log_schema().source_type_key(), Bytes::from("demo_logs"));
                         log.try_insert(log_schema().timestamp_key(), now);
 
-                        out.send(event)
-                            .await
-                            .map_err(|_: crate::source_sender::ClosedError| {
+                        out.send(event).await.map_err(
+                            |error: crate::source_sender::ClosedError| {
+                                emit!(&StreamClosedError { error, count: 1 });
```

Comment:
Is it worth updating the `count` to be the remaining number of events rather than `1`?

Reply:
Oh, I see your point! We could also do a `send_all` instead of sending them one by one 😅